### PR TITLE
feat: Adding barometer to the instrument section

### LINKED ISSUE #1094: Implement Pressure sensor in Instruments with updated UI
**Actual Behaviour**
Barometer should have a new activity as an instrument and the UI needs to be changed accordingly

**Expected Behaviour**
Update the UI of the barometer and make it functional in the instrument section.

**Steps to reproduce it**
None

**LogCat for the issue**
None

**Screenshots of the issue**
barometer-altimeter
barometer-pressure-tracker

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by yatri1609**
Fixes #1094 

**Changes**: 
Added barometer tile in the instrument section and worked half way on the implementation of the barometer

**Screenshot/s for the changes**:
barometer_layout_pslab
barometer

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[barometer.zip](https://github.com/fossasia/pslab-android/files/2194672/barometer.zip)
Kindly test it on your devices using the BMP180 sensor as due to lack of device I'm not able to test it.

### CODE REVIEW

**Comment by yatri1609** [comment]
@CloudyPadmal Please add a WIP tag to this PR

**Comment by abhinavraj23** [comment]
@yatri1609 Why did you closed previous PR and not continue with it? From the next time please continue in the same PR, it is the best practice.

File: app/src/main/res/layout/fragment_barometer.xml [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+            <LinearLayout
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:layout_marginTop="@dimen/lux_card_margin_top"
+                android:orientation="vertical">
+
+                <LinearLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="0dp"
```
Comment (abhinavraj23) [inline app/src/main/res/layout/fragment_barometer.xml]: @yatri1609 Please add the dimensions in dimens file.

File: app/src/main/res/layout/fragment_barometer.xml [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+                        android:gravity="center_vertical"
+                        android:textAlignment="center"
+                        android:textColor="@color/black"
+                        android:textSize="@dimen/text_size" />
+
+                </LinearLayout>
+
+                <LinearLayout
+                    android:layout_width="match_parent"
+                    android:layout_height="0dp"
```
Comment (abhinavraj23) [inline app/src/main/res/layout/fragment_barometer.xml]: @yatri1609 Please add the dimensions in dimens file.

File: app/src/main/res/layout/fragment_barometer.xml [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+                        android:layout_alignParentTop="true"
+                        android:layout_centerHorizontal="true"
+                        android:layout_gravity="center_vertical|center_horizontal|center"
+                        android:foregroundGravity="center_vertical"
+                        android:gravity="center_vertical|center_horizontal|center"
+                        android:text="@string/time"
+                        android:textColor="#ffff"
+                        android:textSize="@dimen/lux_chart_axis_text"
+                        android:textStyle="normal|bold"
+                        tools:layout_editor_absoluteX="288dp"
```
Comment (abhinavraj23) [inline app/src/main/res/layout/fragment_barometer.xml]: Please remove these dimensions it will give error in the code.

File: app/src/main/res/layout/fragment_barometer.xml [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+                        android:id="@+id/tv_graph_label_xaxis_hmc"
+                        android:layout_width="wrap_content"
+                        android:layout_height="wrap_content"
+                        android:layout_alignParentTop="true"
+                        android:layout_centerHorizontal="true"
+                        android:layout_gravity="center_vertical|center_horizontal|center"
+                        android:foregroundGravity="center_vertical"
+                        android:gravity="center_vertical|center_horizontal|center"
+                        android:text="@string/time"
+                        android:textColor="#ffff"
```
Comment (abhinavraj23) [inline app/src/main/res/layout/fragment_barometer.xml]: @yatri1609 Please use standard colours of PSLab

File: app/src/main/res/layout/fragment_barometer.xml [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+                </RelativeLayout>
+
+                <android.support.constraint.ConstraintLayout
+                    android:id="@+id/chart_yaxis_layout_hmc"
+                    android:layout_width="wrap_content"
+                    android:layout_height="match_parent"
+                    android:layout_alignParentLeft="true"
+                    android:layout_alignParentStart="true"
+                    android:layout_alignParentTop="true"
+                    android:background="#000"
```
Comment (abhinavraj23) [inline app/src/main/res/layout/fragment_barometer.xml]: @yatri1609 Also here please use the standard colour of PSLab

**Comment by yatri1609** [comment]
@abhinavraj23 I'll make the changes soon

**Comment by yatri1609** [comment]
@abhinavraj23 When I did as per you instructed me to it automatically got closed so I had to open a new PR !

File: app/src/main/java/org/fossasia/pslab/fragment/BarometerFragment.java [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setRetainInstance(true);
+        currentMin = 10000;
+        entries = new ArrayList<>();
+        switch (sensorType) {
+            case 0:
+                sensorManager = (SensorManager) getContext().getSystemService(SENSOR_SERVICE);
+                sensor = sensorManager != null ? sensorManager.getDefaultSensor(Sensor.TYPE_LIGHT) : null;
```
Comment (Vikum94) [inline app/src/main/java/org/fossasia/pslab/fragment/BarometerFragment.java]: Why do you want the Light sensor?

**Comment by yatri1609** [comment]
@vikum94 I didn't removed the compass instrument and it's resources I just merged the upstream to my development branch... Still I will have a look at it..

File: app/src/main/java/org/fossasia/pslab/activity/MultimeterActivity.java [hunk 7]
```diff
@@ -161,7 +161,7 @@ public void onClick(View view) {
                         capacitanceUnit = "nF";
                     } else if (capacitance < 1e-3) {
                         Capacitance = capacitanceFormat.format((capacitance / 1e-6));
-                        capacitanceUnit = "\u00B5"+"F";
+                        capacitanceUnit = "\u00B5" + "F";
                     } else if (capacitance < 1e-1) {
                         Capacitance = capacitanceFormat.format((capacitance / 1e-3));
                         capacitanceUnit = "mF";
```
Comment (abhinavraj23) [inline app/src/main/java/org/fossasia/pslab/activity/MultimeterActivity.java]: @yatri1609 These changes were done in my PR as said by @harsh-2711 please sync your branch with development branch of upstream try `git reset --soft upstream/development` after you have fetched the upstream.

**Comment by yatri1609** [comment]
I just did as you said @abhinavraj23 and it got closed

**Comment by CloudyPadmal** [comment]
@yatri1609 you may do a soft reset tracking the upstream/development branch and force push

File: app/src/main/res/layout/fragment_barometer.xml [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+                        android:layout_alignParentTop="true"
+                        android:layout_centerHorizontal="true"
+                        android:layout_gravity="center_vertical|center_horizontal|center"
+                        android:foregroundGravity="center_vertical"
+                        android:gravity="center_vertical|center_horizontal|center"
+                        android:text="@string/time"
+                        android:textColor="@color/white"
+                        android:textSize="@dimen/lux_chart_axis_text"
+                        android:textStyle="normal|bold"
+                        tools:layout_editor_absoluteX="@dimen/graph_editor_absoluteX"
```
Comment (abhinavraj23) [inline app/src/main/res/layout/fragment_barometer.xml]: @yatri1609 why are you using absolute X and Y this generally tend to give error in the build and  also the UI doesn't remains stable, please remove it.

File: app/src/main/res/drawable/tile_icon_compass.xml [hunk 9]
```diff
@@ -1,17 +0,0 @@
...
-        android:strokeColor="#ffffff" android:strokeLineCap="butt"
-        android:strokeLineJoin="miter" android:strokeWidth="0.40000001"/>
-    <path android:fillColor="#FFFFFFFF"
-        android:pathData="m7.515,1.411a0.265,0.265 0,0 0,-0.045 0.005c-3.318,0.026 -6.007,2.715 -6.035,6.033a0.265,0.265 0,0 0,0 0.1c0.026,3.318 2.715,6.008 6.033,6.036a0.265,0.265 0,0 0,0.1 0c3.318,-0.026 6.008,-2.715 6.036,-6.033a0.265,0.265 0,0 0,0 -0.1c-0.026,-3.319 -2.717,-6.01 -6.036,-6.036a0.265,0.265 0,0 0,-0.053 -0.005zM7.254,1.95v0.258a0.265,0.265 0,1 0,0.529 0v-0.258c2.861,0.134 5.151,2.424 5.285,5.285h-0.258a0.265,0.265 0,1 0,0 0.529h0.258c-0.134,2.861 -2.424,5.151 -5.285,5.285v-0.258a0.265,0.265 0,0 0,-0.269 -0.268,0.265 0.265,0 0,0 -0.26,0.268v0.258c-2.861,-0.134 -5.151,-2.424 -5.285,-5.285h0.258a0.265,0.265 0,1 0,0 -0.529h-0.258c0.134,-2.861 2.424,-5.151 5.285,-5.285zM10.684,4.06a0.265,0.265 0,0 0,-0.139 0.045l-3.789,2.56a0.265,0.265 0,0 0,-0.071 0.071l-2.56,3.789a0.265,0.265 0,0 0,0.367 0.367l3.79,-2.56a0.265,0.265 0,0 0,0.071 -0.071l2.56,-3.789a0.265,0.265 0,0 0,-0.228 -0.413zM7.094,7.075 L7.943,7.924 5.328,9.691z" android:strokeWidth="0.26458332"/>
-</vector>
```
[the remark this answers is not in the document]
Comment (yatri1609) [reply app/src/main/res/drawable/tile_icon_compass.xml]: Ya I'll have a look into it

**Comment by CloudyPadmal** [comment]
@yatri1609 can you please check why the Travis build is failing?

**Comment by abhinavraj23** [comment]
@yatri1609 have a look at the other tiles, the lines are matching to the circle but in your case it doesn't matches, please do the necessary changes

File: app/src/main/AndroidManifest.xml [hunk 10]
```diff
@@ -81,7 +81,9 @@
 
         <activity android:name=".activity.AccelerometerActivity"/>
 
-        <activity android:name=".activity.MapsActivity" />
```
Comment (abhinavraj23) [inline app/src/main/AndroidManifest.xml]: @yatri1609 Please undo this change as the manifest should contain the maps activity

File: app/src/main/res/values/dimens.xml [hunk 11]
```diff
@@ -303,6 +303,11 @@
     <dimen name="weight_1">1</dimen>
     <dimen name="weight_2">2</dimen>
 
+    <dimen name="barometer_margin_top">54dp</dimen>
+
+    <dimen name="graph_editor_absoluteY">0dp</dimen>
```
Comment (abhinavraj23) [inline app/src/main/res/values/dimens.xml]: @yatri1609 Please remove these dimensions, I don't think they are used in the layout anymore

**Comment by yatri1609** [comment]
Ya I'll make the changes @abhinavraj23

File: app/src/main/res/values/dimens.xml (hunk 11 above)
Comment (yatri1609) [reply app/src/main/res/values/dimens.xml]: Yes I'll remove it

**Comment by abhinavraj23** [comment]
@yatri1609 Please update the screenshot of the tile for barometer

**Comment by yatri1609** [comment]
@abhinavraj23 I have already updated the screenshot in the top comment. Please have a look.

**Comment by yatri1609** [comment]
@abhinavraj23 @Vikum94 @Avjeet @harsh-2711 Can you please review this?

**Comment by mariobehling** [comment]
@yatri1609 

> please complete the bottomsheet guide for barometer

Please follow up on this in the next PR. Thank you!
